video/out/drm_atomic.c

```diff
@@ -458,3 +485,93 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
 
     return (ret == 0);
 }
+
+static inline uint16_t clamp16(float x)
+{
+    return (uint16_t)MPCLAMP(x, 0.0, 65535.0);
+}
+
+static inline uint16_t clamp16_round(float x)
+{
+    return clamp16(roundf(x * 50000.0));
+}
+
+int drm_create_hdr_metadata(struct drm_atomic_context *ctx, struct drm_hdr *hdr_data)
+{
+#if HAVE_DRM_HDR
+    if (mp_colorspace_equal(hdr_data->new_csp, hdr_data->current_csp))
+        return 0;
+
+    if (hdr_data->blob_id != 0)
+        drm_destroy_hdr_metadata(ctx, hdr_data);
```

Comment:
It could be just me, but calling "destroy_foo" at the very start of function called "create_foo" seems counter-intuitive.

Reply:
How about moving the destroy in the caller?